docs/CppInteroperability/Roadmap.md

```diff
@@ -0,0 +1,267 @@
+
+To be able to follow the “Swift API model,” that is to have expressive APIs with strong idioms, the following goals are nessisary:
+
+* C++ interoperability aims to provide a clear, well-defined mapping for APIs that can be imported, and to draw a clear boundary between these APIs and APIs that cannot be imported. A user should be able to read the C++ interoperability documentation and have a good idea of both how much of their C++ API will be able to be imported in Swift and what that API will look like once it is imported. 
+* It is a goal to preserve the semantics (including performance semantics) that an API was written with by using information provided by the user to understand an API's semantics and mapping those semantics to a Swift idiom that represents the same thing.
+* It is *not* a goal to import every C++ API into Swift. APIs that are imported by Swift should feel Swifty and idiomatic. There must be an easy way to add annotations so that APIs become importable, especially when updating APIs directly is not an option (because the source code is not available). 
+
+
+Finally, Swift programmers have used C++ APIs since Swift 1.0 through an Objective-C or C bridging layer. It is important that programmers are able to *incrementally* remove these (potentially huge) bridging layers and start using their C++ APIs directly. To make this transition incremental, C++ interoperability must not change the way Swift imports any existing C or Objective-C APIs, even when C++ interoperability is enabled. 
+
+
+## The approach
+
+The approach for importing C++ APIs into Swift is derived from the fact that both our goals, necessitate that not all APIs are imported automatically; to import an API the user must provide semantic information in some cases. Further, in order to realize our goals (especially the goals of safety and clear, well defined mappings), it is impossible to import any arbitrary C++ API. APIs that are imported by Swift must fall into one of our understood categories or be annotated with semantic information that can be used to classify them into a Swift idiom. 
+
+In this section, the roadmap will discuss several of the most common C++ API patterns and how they might be imported by the Swift compiler using the goals outlined above.
+
+### Reference types
```

Comment:
It may be good to also discuss the possibility of having a "Swift reference type" or a foreign reference type that has Swift reference counting.

Reply:
I would suggest starting with (trivial and non-trivial) value types, since they're the default patterns we're going to try to recognize.

You can explain the value/reference type distinction once and then assume it later; no need to repeat it in different sections.